### PR TITLE
 Automatic sysroot inclusion in macOS

### PR DESCRIPTION
**Pull request by rafaelsamenezes**
Small QoL fixes for macOS

### CODE REVIEW

**Comment by fbrausse** [comment]
In #905 I'm doing something similar, but with a cmake variable (additionally touches Options.cmake and cmake_config.in). We actually already have the `C2GOTO_SYSROOT` which is used in the generated `macos-wrapper.sh`. Would you consider using (and defaulting) that variable instead of the hard-coded path? That way we could even remove generating the `macos-wrapper.sh` (if it isn't required for backwards compat).

**Comment by rafaelsamenezes** [comment]
> Would you consider using (and defaulting) that variable instead of the hard-coded path?

Sure, good idea

**Review by lucasccordeiro — APPROVED** [review]

File: src/clang-c-frontend/clang_c_language.cpp [hunk 1]
```diff
@@ -165,7 +165,14 @@ void clang_c_languaget::build_compiler_args(const std::string &tmp_dir)
 
   if(config.ansi_c.target.is_macos())
   {
+    /* NOTE: clang cannot deduce the sysroot automatically
+     *  we could invoke system's clang and try to find out
+     * the correct place... but that seems an overkill
+     */
+    if(sysroot.empty())
+      compiler_args.emplace_back("--sysroot=" ESBMC_C2GOTO_SYSROOT);
```
Comment (fbrausse) [inline src/clang-c-frontend/clang_c_language.cpp]: I believe this is now independent of `config.ansi_c.target.is_macos()`. Locally, my clang accepts an empty `--sysroot=` parameter without modifying the search paths.

**Review by fbrausse — APPROVED** [review]
LGTM! Thanks for addressing my comments, @rafaelsamenezes.

**Comment by lucasccordeiro** [comment]
Thanks for submitting this PR, @rafaelsamenezes.

**Pull request merged**